src/corelibs/U2Core/src/datatype/msa/MultipleChromatogramAlignmentRow.cpp

```diff
@@ -352,14 +352,44 @@ qint64 MultipleChromatogramAlignmentRowData::getBaseCount(qint64 before) const {
     return MsaRowUtils::getUngappedPosition(gaps, sequence.length(), trimmedRowPos, true);
 }
 
+const QMap<DNAChromatogram::Trace, QVector<ushort> DNAChromatogram::*> PEAKS =
+                    { {DNAChromatogram::Trace::Trace_A, &DNAChromatogram::A},
+                      {DNAChromatogram::Trace::Trace_C, &DNAChromatogram::C},
+                      {DNAChromatogram::Trace::Trace_G, &DNAChromatogram::G},
+                      {DNAChromatogram::Trace::Trace_T, &DNAChromatogram::T} };
+
 QPair<DNAChromatogram::ChromatogramTraceAndValue, DNAChromatogram::ChromatogramTraceAndValue>
-                MultipleChromatogramAlignmentRowData::getTwoHighestPeaks(qint64 position) const {
-    const int baseCall = chromatogram.baseCalls[position];
-    QList<DNAChromatogram::ChromatogramTraceAndValue> peaks =
-                                    { { DNAChromatogram::Trace::Trace_A, chromatogram.A[baseCall] },
-                                      { DNAChromatogram::Trace::Trace_C, chromatogram.C[baseCall] },
-                                      { DNAChromatogram::Trace::Trace_G, chromatogram.G[baseCall] },
-                                      { DNAChromatogram::Trace::Trace_T, chromatogram.T[baseCall] } };
+                MultipleChromatogramAlignmentRowData::getTwoHighestPeaks(qint64 position, bool& hasTwoPeaks) const {
+    hasTwoPeaks = true;
+    int previousBaseCall = chromatogram.baseCalls[position != 0 ? position - 1 : position];
+    int baseCall = chromatogram.baseCalls[position];
+    int nextBaseCall = chromatogram.baseCalls[position != (chromatogram.baseCalls.size() - 1) ? position + 1 : position];
+    QList<DNAChromatogram::ChromatogramTraceAndValue> peaks;
+
+    for (auto peak : PEAKS.keys()) {
+        const auto& chromatogramVector = chromatogram.*PEAKS.value(peak);
```

Comment:
Use type name instead of "auto" if the type name is not clear (not in the same line or in a several lines above of the current)
As a reviewer (aka code reader) I can't understand what is the type name here.

Reply:
OK